src/assembler.cpp

```diff
@@ -443,22 +443,21 @@ void Assembler::handleLabel(std::any const& lbl)
         return;
     }
 
-    std::string label = std::string(std::any_cast<std::string_view>(lbl));
+    std::string label { std::any_cast<std::string_view>(lbl) };
 
     if (label == "$" || label == "-" || label == "+") {
-        if (inMacro != 0) throw parse_error("No special labels in macro");
+        ::Check(inMacro == 0, "No special labels in macro");
         label = "__special_" + std::to_string(labelNum);
         labelNum++;
     } else {
         if (label[0] == '.') {
-            if (lastLabel.empty()) {
-                throw parse_error("Local label without global label");
-            }
+            ::Check(!lastLabel.empty(), "Local label without global label");
             label = std::string(lastLabel) + label;
         } else {
             lastLabel = std::any_cast<std::string_view>(lbl);
         }
     }
+    ::Check(syms.is_defined(label) == false, fmt::format("Symbol '{}' already declared", label));
```

Comment:
Sadly this doesn't work -- any time more than one pass is required this will cause an error.

Reply:
Alright, wasn't aware this is "layered".

Reply:
Removed the commit then.